ReproducibleEnvs.Rmd

```diff
@@ -0,0 +1,564 @@
+
+* `/Projects/user1/project/data/project_data.csv`
+* `/Projects/user2/phs_projects/project/data/project_data.csv`
+
+While less of an issue in PHS as we all use Posit Workbench to access RStudio, there can also be differences introduced depending on operating systems. If you're working with external collaborators for instance, they may use a Windows system which would require different formatting for file paths. 
+
+Taking this a step further, the [`{here}` package](https://here.r-lib.org/) offers up a simple function that protects this way of working even further, and is closely linked with RStudio Projects. The `data/project_data.csv` would be be accessed with `here::here('data', 'project_data.csv')`.
+
+```{r renv-proj, echo=FALSE, fig.align='center'}
+knitr::include_graphics("images/illustrations/r_cracked_setwd.png")
+```
+
+### Environments
+
+Understanding environments in this context is the next fundamental. In the context of using computers to handle and analyse data, to developing outputs such as dashboards and applications, an environment refers to the collection of software tools, libraries, configurations, and dependencies that are required to run a particular program or project. Environments can vary based on factors such as programming language, project requirements, and external dependencies. Also, while we use Posit Workbench in PHS, this eliminates some problems as we're all starting from the same point, but each user can then set their own configuration and packages that will inevitably be different from the next person. Managing these environments effectively is crucial for ensuring that workflows are reproducible across time but also between users and, ultimately, across systems.
+
+```{r renv-env, echo=FALSE, fig.align='center', out.width='100%'}
+knitr::include_graphics("images/renv-positenv.png")
+```
+
+Using Posit Workbench means that our local environment has little impact on any development work. The main difference that can occur between projects and users for R projects is packages, which packages and which version of those packages. That's where `{renv}` comes in, so lets get a look at how to set it up...
+
+## Setting up `{renv}`
+
+The `{renv}` package is a package like any other, so this needs to be installed in order to work with it. This package is available through the PHS Package Manager (or generally through CRAN) and can be installed using `install.packages("renv")`. On PHS Posit Workbench, this will be installed in your user library as standard, this is then used from here to generate the project level library.
```

Comment:
```suggestion
`{renv}` is a package like any other, so must first be installed. It is available through the PHS Package Manager (or generally through CRAN) and can be installed using `install.packages("renv")`. On PHS Posit Workbench, this will be installed in your user library as standard, this is then used to generate the project level library.
```